ctwrap/modules/ignition.py

```diff
@@ -78,33 +81,46 @@ def run(name, chemistry=None, initial=None, simulation=None):
     delta_t = par.delta_t
     n_points = par.n_points
 
-    # preprocess
-
-    # create pandas data frame
-    keys = ['t (s)', 'T (K)', 'rho (kg/m3)'] + \
-        gas.species_names
-    df = pd.DataFrame(columns=keys)
+    # SolutionArray
+    states = ct.SolutionArray(gas, extra=['t'])
 
     # Run simulation
-
     time = sim.time
 
     # loop: note that advance() returns boolean
-    for i in range(par.n_points):
+    for i in range(n_points):
 
         time += delta_t
         sim.advance(time)
 
-        df_len = len(df)
-        row = (time, gas.T, gas.density) + tuple(gas.X)
-        df.loc[df_len] = row
+        states.append(reactor.thermo.state, t=time)
+
+    out = {}
+    out[name] = states
+
+    return out
+
 
-    return {name: df}
+def save(filename, data, task=None):
+    """
+    This function saves the output from the run method
+
+    Arguments:
+        filename (str): naming of file
+        data (Dict): data to be saved
+        task (str): name of task if running variations
+    """
+    attrs = {}
 
+    attrs['description'] = task
+    for group, solution_array in data.items():
```

Comment:
I believe there's only one entry in the dictionary, right? Also, use `states` instead of `solution_array` to reflect code above.